tests/gtest/test_netif.cpp

```diff
@@ -297,4 +360,65 @@ TEST(Netif, WpanIfHasCorrectUnicastAddresses_AfterUpdatingUnicastAddresses)
+    otbr::Netif netif;
+    EXPECT_EQ(netif.Init(wpan), OT_ERROR_NONE);
+
+    otbr::Ip6Address kDefaultMulAddr1 = {
+        {0xff, 0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01}};
+    const char *kDefaultMulAddr1Str = "ff01::1";
+    const char *kDefaultMulAddr2Str = "ff02::1";
+    const char *kDefaultMulAddr3Str = "ff02::2";
+
+    otbr::Ip6Address kMulAddr1 = {
+        {0xff, 0x03, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0xfc}};
+    otbr::Ip6Address kMulAddr2 = {
+        {0xff, 0x32, 0x00, 0x40, 0xfd, 0x0d, 0x07, 0xfc, 0xa1, 0xb9, 0xf0, 0x50, 0x00, 0x00, 0x00, 0x01}};
+    const char *kMulAddr1Str = "ff03::fc";
+    const char *kMulAddr2Str = "ff32:40:fd0d:7fc:a1b9:f050:0:1";
+
+    otbr::Ip6Address testArray1[] = {
+        kMulAddr1,
+    };
+    std::vector<otbr::Ip6Address> testVec1(testArray1, testArray1 + sizeof(testArray1) / sizeof(otbr::Ip6Address));
+    netif.UpdateIp6MulticastAddresses(testVec1);
+    std::vector<std::string> wpanMulAddrs = GetAllIp6MulAddrs(wpan);
+    EXPECT_EQ(wpanMulAddrs.size(), 4);
+    EXPECT_THAT(wpanMulAddrs, ::testing::Contains(kMulAddr1Str));
+    EXPECT_THAT(wpanMulAddrs, ::testing::Contains(kDefaultMulAddr1Str));
+    EXPECT_THAT(wpanMulAddrs, ::testing::Contains(kDefaultMulAddr2Str));
+    EXPECT_THAT(wpanMulAddrs, ::testing::Contains(kDefaultMulAddr3Str));
+
+    otbr::Ip6Address              testArray2[] = {kMulAddr1, kMulAddr2};
+    std::vector<otbr::Ip6Address> testVec2(testArray2, testArray2 + sizeof(testArray2) / sizeof(otbr::Ip6Address));
+    netif.UpdateIp6MulticastAddresses(testVec2);
+    wpanMulAddrs = GetAllIp6MulAddrs(wpan);
+    EXPECT_EQ(wpanMulAddrs.size(), 5);
+    EXPECT_THAT(wpanMulAddrs, ::testing::Contains(kMulAddr1Str));
+    EXPECT_THAT(wpanMulAddrs, ::testing::Contains(kMulAddr2Str));
+    EXPECT_THAT(wpanMulAddrs, ::testing::Contains(kDefaultMulAddr1Str));
+    EXPECT_THAT(wpanMulAddrs, ::testing::Contains(kDefaultMulAddr2Str));
+    EXPECT_THAT(wpanMulAddrs, ::testing::Contains(kDefaultMulAddr3Str));
+
+    std::vector<otbr::Ip6Address> empty;
+    netif.UpdateIp6MulticastAddresses(empty);
+    wpanMulAddrs = GetAllIp6MulAddrs(wpan);
+    EXPECT_EQ(wpanMulAddrs.size(), 3);
+    EXPECT_THAT(wpanMulAddrs, ::testing::Contains(kDefaultMulAddr1Str));
+    EXPECT_THAT(wpanMulAddrs, ::testing::Contains(kDefaultMulAddr2Str));
+    EXPECT_THAT(wpanMulAddrs, ::testing::Contains(kDefaultMulAddr3Str));
+
+    otbr::Ip6Address              testArray3[] = {kDefaultMulAddr1};
+    std::vector<otbr::Ip6Address> testVec3(testArray3, testArray3 + sizeof(testArray3) / sizeof(otbr::Ip6Address));
+    netif.UpdateIp6MulticastAddresses(testVec3);
+    EXPECT_EQ(wpanMulAddrs.size(), 3);
+    EXPECT_THAT(wpanMulAddrs, ::testing::Contains(kDefaultMulAddr1Str));
+    EXPECT_THAT(wpanMulAddrs, ::testing::Contains(kDefaultMulAddr2Str));
+    EXPECT_THAT(wpanMulAddrs, ::testing::Contains(kDefaultMulAddr3Str));
```

Comment:
Add another update to empty list to see the kDefaultMulAddr1 is still subscribed?

Reply:
Sounds good! I moved updating to empty to the end and verify `kDefaultMulAddr1`is still subscribed. This is equivalent.